app/controllers/OverviewController.scala

```diff
@@ -233,29 +253,43 @@ class OverviewController @Inject()(
         .addingToSession((Constants.SRN, srn.value))
     }
 
-  def onSelectStart(srn: Srn, taxYear: String, version: String, reportType: String): Action[AnyContent] =
-    identify.andThen(allowAccess(srn)).andThen(getData).andThen(createData).async { implicit request =>
-      reportType match {
-        case PsrReportType.Sipp.name =>
-          val sippUrl = s"${config.urls.sippBaseUrl}/${srn.value}${config.urls.sippStartJourney}"
-          Future.successful(
-            Redirect(sippUrl)
-              .addingToSession(Constants.TAX_YEAR -> taxYear)
-              .addingToSession(Constants.VERSION -> version)
-          )
-        case _ =>
-          val yearFrom = LocalDate.parse(taxYear)
-          val yearTo = yearFrom.plusYears(1).minusDays(1)
-          val dateRange = DateRange(yearFrom, yearTo)
-          for {
-            userAnswers <- Future.fromTry(request.userAnswers.set(WhichTaxYearPage(srn), dateRange))
-            _ <- saveService.save(userAnswers.copy(id = UNCHANGED_SESSION_PREFIX + userAnswers.id))
-            _ <- saveService.save(userAnswers)
-          } yield {
-            Redirect(controllers.routes.WhatYouWillNeedController.onPageLoad(srn, "", taxYear, version))
-          }
+  def onSelectStart(
+    srn: Srn,
+    taxYear: String,
+    version: String,
+    reportType: String,
+    lastSubmittedPsrFbInPreviousYears: Option[String]
+  ): Action[AnyContent] =
+    identify
+      .andThen(allowAccess(srn))
+      .andThen(createData)
+      .andThen(
+        prePopData(
+          Option.when(reportType == PsrReportType.Standard.name)(lastSubmittedPsrFbInPreviousYears).flatten
+        )
+      )
+      .async { implicit request =>
+        reportType match {
+          case PsrReportType.Sipp.name =>
+            val sippUrl = s"${config.urls.sippBaseUrl}/${srn.value}${config.urls.sippStartJourney}"
+            Future.successful(
+              Redirect(sippUrl)
+                .addingToSession(Constants.TAX_YEAR -> taxYear)
+                .addingToSession(Constants.VERSION -> version)
+            )
+          case _ =>
+            val yearFrom = LocalDate.parse(taxYear)
+            val yearTo = yearFrom.plusYears(1).minusDays(1)
+            val dateRange = DateRange(yearFrom, yearTo)
+            for {
+              userAnswers <- Future.fromTry(request.userAnswers.set(WhichTaxYearPage(srn), dateRange))
+              _ <- saveService.save(userAnswers.copy(id = UNCHANGED_SESSION_PREFIX + userAnswers.id))
+              _ <- saveService.save(userAnswers)
+            } yield {
+              Redirect(controllers.routes.WhatYouWillNeedController.onPageLoad(srn, "", taxYear, version))
```

Comment:
We might need to conditionally redirect to a different URL here in case of pre-population. Do we have the information that the pre-population has just happened?

Reply:
We can understand it if `lastSubmittedPsrFbInPreviousYears` is not None. That means we are going to read prepop data from etmp.

I separated the logic from SIPP. Do you think it is correct ?

Reply:
I think it is ok If we can conditionally redirect and use this later on in the UI flow to distinguish pro-population.

Reply:
We discussed with @cyeniceri and agreed to have a flag saved somewhere (e.g. request or mongo) to indicate that this return is a pre-population.
The value of this flag would be populated on "Start" and on "Continue" link too.
This value could be re-used later on for a conditional navigation and/or the task list/cya pages to dictate if this is a pre-population. FYI @westwater @ccatosdevelopment @narendrapaduchuri .

Reply:
Related: https://github.com/hmrc/pension-scheme-return-frontend/pull/930#discussion_r1810634372